lib/stringify.cpp

```diff
@@ -80,7 +80,9 @@ cstring vprintf_format(const char* fmt_str, va_list ap) {
     if (static_cast<size_t>(size) >= sizeof(buf)) {
         char* formatted = new char[size + 1];
         vsnprintf(formatted, size + 1, fmt_str, ap_copy);
-        return cstring(formatted);
+        cstring result(formatted);
```

Comment:
If you plan to fix all the leaks in the compiler you have a few thousands left to do.
We actually do not use "delete", this code is written using garbage-collected C++.

Reply:
Thank you for review! Since p4c has option to build without GC and in bitvec utility invokes delete [] explicitly too (https://github.com/p4lang/p4c/blob/master/lib/bitvec.cpp#L57), can we at least partially fix memory leaks in build without GC? Is delete keyword no-op for build with GC or this is a bug  (I don't see any magic for GC like here https://github.com/p4lang/p4c/blob/master/lib/indent.cpp#L45) in bitvec?

Reply:
Calling delete explicitly should be fine with the garbage collector (its generally a noop unless the memory was marked non-collectable for some reason -- it makes it collectable).  It does explicitly call the destructor if that is relevant (the garbage collector never calls destructors -- only finalizers)

Reply:
Oh, I found magic here https://github.com/p4lang/p4c/blob/5d83957346aef6d0f1abb3f65691744a4738cc3f/lib/bitvec.h#L424 sorry.
So I can rewrite allocation like here and this will work for both builds?

Reply:
The bitvec code was carefully written to work with or without the garbage collector, as it is used in other projects that don't use the garbage collector.

Reply:
There is no harm in calling delete if it is safe. But don't expect this to be the only leak.

Reply:
Thank you for explanation! Looks like there no more leaks which not require changing API or behavior to fix them, so other leaks out of the scope for this request with minor changes.

There any special limitations for this part? I mean, can we use supported by gcc 4.9 and clang 3.3 C++11 features like magic statics and move semantic?

Reply:
We can use standard c++11 features.
We do want this code to be as portable as possible.